get-armory-cli

```diff
@@ -25,6 +36,23 @@ case $RAW_ARCH in
       ;;
 esac
 
+# Sanity checks
+
+echo "Looking for dependencies..."
+if ! command -v curl > /dev/null; then
+	echo "######################################################################################################"
+	echo "# curl not found."
+	echo "#"
+	echo "# Please install curl on your system using your favourite package manager."
+	echo "######################################################################################################"
+	exit 1
+fi
+
+status_code=$(curl --write-out %{http_code} --silent --output /dev/null https://api.github.com/repos/armory/avm/releases/latest)
+if [[ "$status_code" -eq 403 ]] ; then
+    throw_error_message "GitHub API rate limit exceeded - Check out GitHub documentation for more details."
```

Comment:
pehraps we should:
- log information about rate limiting
- pause for a few seconds
- retry
- repeat like 5 times?

Reply:
Yes, totally agree, the problem I see with the retry is that it could take up to 60 minutes to restart the rate-limit counter, also GitHub documentation mentioned that we shouldn't retry before the time specified by the x-ratelimit-reset header.
https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#exceeding-the-rate-limit

It would be ok if I just print log information about when the user can retry, along with the link above to reference about GitHub rate-limiting?